### PR TITLE
Fix build/distcheck; enable more warnings more often

### PR DESCRIPTION
**Pull request by wjt**
In #224 I forgot to commit one file, and introduced three test failures when running `make distcheck`. Mea culpa. While fixing these, I fixed an existing error when running `make distcheck` as non-root, enabled many more compiler warnings, and fixed them too.

https://phabricator.endlessm.com/T19953

### CODE REVIEW

File: tests/daemon/mock-cache-size-provider.c [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
+
+/* Copyright 2017 Endless Mobile, Inc. */
```
Comment (smspillaz) [inline tests/daemon/mock-cache-size-provider.c]: Probably needs an actual licence header, though if this is the style in the rest of the code, maybe not worth worrying about.

Comment (smspillaz) [reply tests/daemon/mock-cache-size-provider.c]: Oh, derp, its literally 3 lines below. Too hasty to comment!

File: tests/daemon/mock-persistent-cache.c [hunk 2]
```diff
@@ -88,6 +88,7 @@ emer_persistent_cache_new (const gchar *directory,
   return self;
 }
 
+G_GNUC_NORETURN
 EmerPersistentCache *
 emer_persistent_cache_new_full (const gchar              *directory,
                                 guint64                   cache_size,
```
Comment (smspillaz) [inline tests/daemon/mock-persistent-cache.c]: Not sure if this is a good candidate for `G_GNUC_NORETURN`, `g_assert_not_reached` is supposed to be inert when compiled in release mode.

Maybe just `return NULL;` afterwards?

**Comment by smspillaz** [comment]
Just one comment.

File: tests/daemon/mock-persistent-cache.c (hunk 2 above)
Comment (wjt) [reply tests/daemon/mock-persistent-cache.c]: Great point. I'll replace this with `g_return_val_if_reached (NULL);` which always returns the value, even in `G_DISABLE_CHECKS` mode.

File: Makefile.am [hunk 3]
```diff
@@ -30,13 +30,14 @@ ACLOCAL_AMFLAGS = -I m4
...
-AM_CFLAGS = @STRICT_CFLAGS@
+AM_CFLAGS = @WARN_CFLAGS@
+AM_LDFLAGS = @WARN_LDFLAGS@
 
 # Enable strict compiler flags and install systemd unit files to the specified
 # directory when doing 'make distcheck'.
 AM_DISTCHECK_CONFIGURE_FLAGS = \
-	--enable-strict-flags \
-	--with-systemdsystemunitdir=$(libdir)/systemd \
+	--enable-compile-warnings=error \
```
Comment (wjt) [inline Makefile.am]: I also missed  `--enable-strict-flags` here before.

**Pull request merged**